File: Xamarin/mySkatemap.TrackRecorderApp/Infrastructure/Location/ILocationProvider.cs
using System;

namespace mySkatemap.Apps.TrackRecorder.Infrastructure.Location
{
	internal interface ILocationProvider : IDisposable, IObservable<Location>
	{
		void StartLocationUpdates();

		void StopLocationUpdates();

		void RequestLocation();

		Boolean HasRequestedLocationUpdates { get; }

		IObservable<Location> LocationUpdates { get; }
	}
}

File: Android/app/src/main/java/com/janhafner/myskatemap/apps/trackrecorder/services/trackrecorder/provider/ILocationProvider.kt
package com.janhafner.myskatemap.apps.trackrecorder.services.trackrecorder.provider

import com.janhafner.myskatemap.apps.trackrecorder.core.IDestroyable
import com.janhafner.myskatemap.apps.trackrecorder.core.types.Location
import io.reactivex.Observable

internal interface ILocationProvider : IDestroyable {
    fun startLocationUpdates()

    fun stopLocationUpdates()

    val isActive: Boolean

    val locationsReceived: Observable<Location>
}